python/Ganga/Core/GangaRepository/Registry.py

```diff
@@ -259,33 +272,26 @@ class Registry(object):
     Base class providing a dict-like locked and lazy-loading interface to a Ganga repository
     """
 
-    def __init__(self, name, doc, update_index_time=30):
+    def __init__(self, name, doc):
         """Registry constructor, giving public name and documentation
         Args:
             name (str): Name of teh registry e.g. 'jobs', 'box', 'prep'
             doc (str): This is the doc string of the registry describing it's use and contents
-            update_index_time (int): This is used to determine how long to wait between updating the index in the repo
         """
         self.name = name
         self.doc = doc
         self._hasStarted = False
-        self.update_index_time = update_index_time
-        self._update_index_timer = 0
         self._needs_metadata = False
         self.metadata = None
-        self._lock = threading.RLock()
-        self.hard_lock = {}
-        self.changed_ids = {}
+        self._read_lock = threading.RLock()
+        self._flush_lock = threading.RLock()
```

Comment:
... I would prefer to see that these locks are being tested in some form for deadlocks but I suppose it's unlikely not to be caught in our current tests.